cmd/kubeadm/app/util/runtime/runtime_unix.go

```diff
@@ -20,19 +20,29 @@ package util
 
 import (
 	"os"
+	"net"
+	"net/url"
 )
 
 const (
-	dockerSocket     = "/var/run/docker.sock" // The Docker socket is not CRI compatible
-	containerdSocket = "/run/containerd/containerd.sock"
+	dockerSocket     = "unix:///var/run/docker.sock" // The Docker socket is not CRI compatible
+	containerdSocket = "unix:///run/containerd/containerd.sock"
 )
 
 // isExistingSocket checks if path exists and is domain socket
 func isExistingSocket(path string) bool {
-	fileInfo, err := os.Stat(path)
+	u, err := url.Parse(path)
 	if err != nil {
 		return false
 	}
-
-	return fileInfo.Mode()&os.ModeSocket != 0
+	// remove it later if we don't support the path anymore
+	if u.Scheme == "" {
+		u.Scheme = "unix"
```

Comment:
IMO, adding a warning here may be better.

Reply:
please see: https://github.com/kubernetes/kubeadm/issues/2426

Reply:
Updated per your comments.

For the upgrading case, I need to do some more investigation.